src/librustc_mir/borrow_check/diagnostics/conflict_errors.rs

```diff
@@ -691,7 +691,7 @@ impl<'cx, 'tcx> MirBorrowckCtxt<'cx, 'tcx> {
         let root_place_projection = self.infcx.tcx.intern_place_elems(root_place.projection);
 
         if self.access_place_error_reported.contains(&(
-            Place { local: root_place.local.clone(), projection: root_place_projection },
+            Place { local: *root_place.local, projection: root_place_projection },
```

Comment:
cc @matthewjasper Why is there a reference to a `Local` here - is it a mutable reference perhaps?